crates/mysten-metrics/src/lib.rs

```diff
@@ -135,3 +138,29 @@ pub fn monitored_scope(name: &'static str) -> Option<MonitoredScopeGuard> {
         None
     }
 }
+
+pub trait ScopeFutureExt: Future + Sized {
+    fn in_scope(self, name: &'static str) -> ScopeFuture<Self>;
```

Comment:
Can we name this `in_monitored_scope` or something more appropriate, so readers can see its purpose easily?